### PR TITLE
refactor(server+client): update ai user prompt and results to client

### PR DESCRIPTION
**Pull request by abrl91**
Close: https://github.com/amplication/private-issues/issues/141

## PR Details

- update ai user prompt based on the requirements
- update the result based on the requirements
- update modles and client
- update tests

## PR Checklist

- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/resource/resourceBtm.service.ts [hunk 1]
```diff
@@ -178,26 +156,27 @@ export class ResourceBtmService {
...
-      acc[entity.id] = entity.name;
-      return acc;
-    });
+    const entityNameToRelatedFieldsMap = resource.entities.reduce(
+      (acc, entity) => {
+        const relationFields = entity.versions[0].fields.filter(
+          (field) => field.dataType === EnumDataType.Lookup
+        );
+        const entityFields = relationFields.length
+          ? [...new Set(relationFields.map((field) => field.name))]
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/resource/resourceBtm.service.ts:165]: Just one question here about mapping to `field.name`  :
What if the field name is not the name of the remote relation? E.g. I have a field named "manager" that is a relation to the "Employee" table?

Comment (abrl91) [reply packages/amplication-server/src/core/resource/resourceBtm.service.ts:165]: good one!

File: packages/amplication-server/src/core/resource/resourceBtm.service.ts [hunk 2]
```diff
@@ -176,26 +155,41 @@ export class ResourceBtmService {
...
-      acc[entity.id] = entity.name;
-      return acc;
-    });
+    const entityNameToRelatedFieldsMap = resource.entities.reduce(
+      (acc, entity) => {
+        const relationFields = entity.versions[0].fields.filter(
+          (field) => field.dataType === EnumDataType.Lookup
+        );
+
+        const relatedEntityFieldNames = relationFields.length
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/resource/resourceBtm.service.ts:164]: This isn't `relatedEntityFieldNames` but more of `relatedEntities` or `relatedEntittiesNames`

**Review by mulygottlieb — APPROVED** [review]
Approved with one comment regarding variable naming for your consideration

File: packages/amplication-server/src/core/resource/resourceBtm.service.ts (hunk 2 above)
Comment (abrl91) [reply packages/amplication-server/src/core/resource/resourceBtm.service.ts:164]: @mulygottlieb 
Have a look at the return value. It's the names, `string[]`

Comment (abrl91) [reply packages/amplication-server/src/core/resource/resourceBtm.service.ts:164]: or just and empty array

Comment (abrl91) [reply packages/amplication-server/src/core/resource/resourceBtm.service.ts:164]: image

Comment (mulygottlieb) [reply packages/amplication-server/src/core/resource/resourceBtm.service.ts:164]: @abrl91 yeah, names, but entity names and not field names. nevermind :)

**Pull request merged**